### PR TITLE
Add support for HEIC format

### PR DESCRIPTION
**Pull request by stefanoverna**
Hi! Is HEIC format detection interesting to you? :wink:

### CODE REVIEW

**Comment by sdsykes** [comment]
Awesome. I'll review this as soon as I can

**Comment by sdsykes** [comment]
Just a thought, not related to this PR directly, but as it adds more than 150 lines of code - do you think it's time I split fastimage.rb into separate files?

**Comment by stefanoverna** [comment]
Of course it would make sense!

File: lib/fastimage.rb [hunk 1]
```diff
@@ -568,6 +583,153 @@ def parse_size_for_ico
...
+
+    private
+
+    def read_boxes!(max_read_bytes = nil)
+      end_pos = max_read_bytes.nil? ? nil : @stream.pos + max_read_bytes
+      index = 0
+
+      loop do
+        end_pos.nil? || @stream.pos < end_pos or
+          return
```
Comment (sdsykes) [inline lib/fastimage.rb:613]: One thing: I find this a bit hard to read.
It's mostly a matter of taste, but I find the postfix use of 'or' harder to read than an 'if'

    return if end_pos && @stream.pos >= end_pos

Likewise, I prefer this to using 'or' even in the simple case: this is clearer to me

    throw :finish if box_size < 12

**Comment by sdsykes** [comment]
This looks good, I think it's pretty much good to merge. After merge it'll be time to look at dividing into smaller files :)

**Comment by stefanoverna** [comment]
Done @sdsykes, thanks for the review!

**Pull request merged**